### PR TITLE
Visitor can register account

### PR DESCRIPTION
**Pull request by FacundoOs**
# [PT](https://www.pivotaltracker.com/story/show/175113676)

## This Pr contains:
- Acceptance test
- Sign up component
- Add route for sign up
- Post request for sign up

### CODE REVIEW

File: src/components/SignUpForm.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      const password_confirmation = event.target.passwordConfirmation.value;
+
+      const response = await auth.signUp({
+        name: name,
+        email: email,
+        password: password,
+        password_confirmation: password_confirmation,
+      });
+      debugger
+      history.replace("/login", { message: response.data.status });
```
Comment (jennymalmefeldt) [inline src/components/SignUpForm.jsx]: debugger

File: cypress/integration/vistorCanRegisterAnAccount.feature.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      cy.get('[data-cy="sign-up-button"]').contains("Sign Up").click();
+
+      cy.get('[data-cy="sign-up-form"]').within(() => {
+        cy.get('[data-cy="name"]').type("Facundo Osores");
+        cy.get('[data-cy="email"]').type("registered@mail.com");
+        cy.get('[data-cy="password"]').type("password");
+        cy.get('[data-cy="password-confirmation"]').type("wrongPassword");
+        cy.get('[data-cy="submit"]').contains("Submit").click();
+      });
+    });
```
Comment (ctl-lab) [inline cypress/integration/vistorCanRegisterAnAccount.feature.js]: there is no assertion in this it block.

File: cypress/integration/vistorCanRegisterAnAccount.feature.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      cy.get('[data-cy="sign-up-form"]').within(() => {
+        cy.get('[data-cy="name"]').type("Facundo Osores");
+        cy.get('[data-cy="email"]').type("registered@mail.com");
+        cy.get('[data-cy="password"]').type("password");
+        cy.get('[data-cy="password-confirmation"]').type("password");
+        cy.get('[data-cy="submit"]').contains("Submit").click();
+      });
+    });
+  });
+});
```
Comment (ctl-lab) [inline cypress/integration/vistorCanRegisterAnAccount.feature.js]: missing assertion

File: src/components/LoginForm.jsx [hunk 4]
```diff
@@ -26,12 +28,23 @@ const LoginForm = () => {
 
       history.replace({ pathname: "/" });
     } catch (error) {
-      setmessage(error.response.data.errors[0]);
+      setMessage(error.response.data.errors[0]);
```
Comment (ctl-lab) [inline src/components/LoginForm.jsx:32]: good catch!

File: src/components/SignUpForm.jsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    event.preventDefault();
+    try {
+      const name = event.target.name.value;
+      const email = event.target.email.value;
+      const password = event.target.password.value;
+      const password_confirmation = event.target.passwordConfirmation.value;
+
+      const response = await auth.signUp({
+        name: name,
+        email: email,
```
Comment (ctl-lab) [inline src/components/SignUpForm.jsx:17]: why to send it as an object?

Comment (ctl-lab) [reply src/components/SignUpForm.jsx:17]: this one was just to know. No need to change it.

Comment (FacundoOs) [reply src/components/SignUpForm.jsx:17]: This is how we have to send the parameters to the backend for sign up

**Pull request merged**